File: Data Structures/C/LinkedList.c
#include <stdio.h>
#include <stdlib.h>
#include "LinkedList.h"

void add(struct LinkedList** head, int item)
{
     struct LinkedList* node = (struct LinkedList*) malloc(sizeof(struct LinkedList*));

     if(node == NULL)
     {
          fprintf(stderr, "Unable to allocate memory for new node\n");
          exit(-1);
     }

     node->item = item;
     node->next = NULL;

     if(*head == NULL)
     {
          *head = node;
     } else {

        struct LinkedList *curr = *head;
        while (curr->next != NULL)
        {
            curr = curr->next;
        }

        curr->next = node;
    }
}

void printList(struct LinkedList* head)
{
     struct LinkedList* point = head;
     while(point != NULL)
     {
          printf("%d ", point->item);
          point = point->next;
     }

}

void freeList(struct LinkedList** head)
{
     struct LinkedList* temp;

     while(*head != NULL)
     {
          temp = *head;
          *head = (*head)->next;
          free(temp);
     }
}

File: Data Structures/Python/LinkedList.py
class Node:

    def __init__(self, i):
        self.item = i
        self.next = None

class LinkedList:

    def __init__(self):
        self.head = None
        self.size = 0

    def add(self, item):
        n = Node(item)
        curr = self.head

        if self.size == 0:
            self.head = Node(item)
            self.size = self.size + 1

        else:
            while curr.next != None:
                curr = curr.next

            curr.next = n
            self.size = self.size + 1

    def printList(self):
        curr = self.head

        while curr != None:
            print(str(curr.item), end=' ')
            curr = curr.next



def LinkedListTest():
    ll = LinkedList()

    ll.add(1)
    ll.add(2)
    ll.add(3)

    #1 2 3
    ll.printList()

LinkedListTest()
